src/fmu/tools/qcforward/_blockedwells_vs_gridprops.py

```diff
@@ -154,29 +158,41 @@ def run(
             dfr, "blocked wells vs grid props", stopaction=False
         )
 
-        # make it possible to print the full dataframe, either some (.e.g the failing)
-        # or all
-        if self.ldata.show_data:
-            if "fail" in self.ldata.show_data and status == "STOP":
-                dfr_stop = dfr[dfr["STATUS"] == "STOP"]
-                wells = list(dfr_stop["WELL"])
-                if "all" in wells:
-                    wells.remove("all")
-                print(f"Failing wells are these {wells}:")
-                print(comb[comb["WELLNAME"].isin(wells)].to_string())
-            elif "warn" in self.ldata.show_data:
-                dfr_warn_or_stop = dfr[dfr["STATUS"] != "OK"]
-                wells = list(dfr_warn_or_stop["WELL"])
-                if "all" in wells:
-                    wells.remove("all")
+        # make it possible to print the underlying dataframe, either some wells (.e.g
+        # the failing) or all wells. If 'fail' it will only show those lines that
+        # contains FAIL
+        show = self.ldata.show_data
+        if show and isinstance(show, dict):
+            if (
+                not isinstance(show, dict)
+                and "lines" not in show
+                and "wellstatus" not in show
+            ):
```

Comment:
```suggestion
        if show is not None:
            if (
                not isinstance(show, dict)
                or ("lines" not in show)
                or ("wellstatus" not in show)
            ):
```

This Error will never happen since you are checking that show is on dictionary form on the first line. Maybe this is a better solution?

Reply:
`if isinstance(show, dict) and show:` I think will do it